reactivesocket-cpp/src/ConnectionAutomaton.h

```diff
@@ -36,7 +37,10 @@ class ConnectionAutomaton :
     /// Registered as an input in the DuplexConnection.
     public Subscriber<Payload>,
     /// Receives signals about connection writability.
-    public Subscription {
+    public Subscription,
+    /// The ConnectionAutomaton instance is shared between streams and also
+    /// ReactiveSocket class, using refcounting for lifetime management
+    public IntrusiveDeleter {
```

Comment:
Ugh, you really don't want everyone to call increment/decrementRefCount, please make it protected inheritance.